hubblestack/extmods/modules/conf_publisher.py

```diff
@@ -0,0 +1,46 @@
+
+import logging
+import hubblestack.splunklogging
+import copy
+import time
+
+log = logging.getLogger(__name__)
+
+def publish():
+    log.info('Started publishing config to splunk')
+    
+    initialize_splunklogging()
+    filtered_conf=filter_config()
+    class MockRecord(object):
+            def __init__(self, message, levelname, asctime, name):
+                self.message = message
+                self.levelname = levelname
+                self.asctime = asctime
+                self.name = name
+
+    handler = hubblestack.splunklogging.SplunkHandler()
+    handler.emit(MockRecord(filtered_conf, 'INFO', time.asctime(), 'hubblestack.hubble_config'))
+    log.info('Published config to splunk')	
+
+def initialize_splunklogging():
+    hubblestack.splunklogging.__grains__ = __grains__
+    hubblestack.splunklogging.__salt__ = __salt__
+
+#Filter out keys containing certain patterns to avoid sensitive information being sent to splunk
```

Comment:
These comments could be moved into docstrings for the function -- that's the accepted standard for python.